apps/web/src/components/transactions/TxDetails/Summary/SafeTxHashDataRow/index.tsx

```diff
@@ -1,28 +1,39 @@
+import { useMemo } from 'react'
 import { TxDataRow, generateDataRowValue } from '../TxDataRow'
 import { type SafeTransactionData, type SafeVersion } from '@safe-global/safe-core-sdk-types'
-import useSafeAddress from '@/hooks/useSafeAddress'
+import { calculateSafeTransactionHash } from '@safe-global/protocol-kit/dist/src/utils'
+import useSafeInfo from '@/hooks/useSafeInfo'
 import useChainId from '@/hooks/useChainId'
 import { getDomainHash, getSafeTxMessageHash } from '@/utils/safe-hashes'
 
 export const SafeTxHashDataRow = ({
-  safeTxHash,
   safeTxData,
-  safeVersion,
+  safeTxHash,
 }: {
-  safeTxHash: string
-  safeTxData?: SafeTransactionData
-  safeVersion: SafeVersion
+  safeTxData: SafeTransactionData
+  safeTxHash?: string
 }) => {
+  const { safe, safeAddress } = useSafeInfo()
   const chainId = useChainId()
-  const safeAddress = useSafeAddress()
+  const safeVersion = safe.version as SafeVersion
+
+  const computedHash = useMemo(() => {
+    if (safeTxHash) return safeTxHash
+    if (!safe.version) return
+    try {
+      return calculateSafeTransactionHash(safeAddress, safeTxData, safe.version, BigInt(safe.chainId))
```

Comment:
If we are rendering a SafeTx in the history of a Safe this could lead to wrong hashes if the Safe has been upgraded from e.g. 1.1.0 to 1.3.0 / 1.4.1 as the domain separator changed.

Considering that we recompute and validate the `safeTxHash` anyways in `useValidateTxData` in the signing modals I would use the given `safeTxHash`

Reply:
It doesn't exist when creating a new tx. This branch of code runs for this case.

Reply:
Oh I am blind. Never mind! I missed the early return -.-